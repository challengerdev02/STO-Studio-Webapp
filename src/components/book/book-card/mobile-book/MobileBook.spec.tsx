import { cleanup, render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import { MobileBook } from './index';

describe('MobileBook component', function () {
  afterEach(cleanup);
  let props = {
    title: 'This is a test title',
    author: 'Cosmas',
    description: 'Meta NFT',
    imageUrl: '/assets/mobile-book-img.svg',
  };

  const setup = (extraProps: Record<string, any> = {}) => {
    const utils = render(
      <MobileBook simulateMobile {...props} {...extraProps} />
    );
    return {
      ...utils,
    };
  };

  beforeEach(() => {
    Object.defineProperty(window, 'matchMedia', {
      writable: true,
      value: jest.fn().mockImplementation((query) => ({
        matches: false,
        media: query,
        onchange: null,
        addListener: jest.fn(), // deprecated
        removeListener: jest.fn(), // deprecated
        addEventListener: jest.fn(),
        removeEventListener: jest.fn(),
        dispatchEvent: jest.fn(),
      })),
    });
  });

  it('should test the MobileBook component has an image displayed', () => {
    const { getByTestId } = setup();
    expect(getByTestId('img')).toBeInTheDocument();
  });

  it('should test the MobileBook component has the necessary texts been displayed ', () => {
    const { getByTestId, getByRole } = setup();
    const button = getByRole('button', { name: /sell/i });
    expect(getByTestId('author')).toBeInTheDocument();
    expect(getByTestId('mobile-book-title')).toBeInTheDocument();
    expect(button).toBeInTheDocument();

    expect(getByTestId('author').textContent).toBe('by @Cosmas');
    expect(getByTestId('mobile-book-title').textContent).toBe(
      'This is a test title'
    );
  });

  it('should match with snapshot', () => {
    // @ts-ignore
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});
