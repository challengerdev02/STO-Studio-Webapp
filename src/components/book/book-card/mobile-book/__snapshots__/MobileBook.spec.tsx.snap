// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileBook component should match with snapshot 1`] = `
<div>
  <div
    class="ant-card sc-bdvvtL EjgXq"
  >
    <div
      class="ant-card-body"
    >
      <div
        class="sc-gKclnd fBXsEO"
      >
        <div
          style="width: 100%;"
        >
          <div
            class="sc-gsDKAQ eeJeSz"
          >
            <img
              class="sc-dkPtRN bRdbyx"
              data-testid="img"
              src="/assets/mobile-book-img.svg"
            />
          </div>
        </div>
        <div>
          <div
            class="sc-hKwDye jyZsmS"
            data-testid="mobile-book-title"
          >
            This is a test title
          </div>
          <div
            class="sc-eCImPb bwuyDl"
            data-testid="author"
          >
            by @
            Cosmas
          </div>
          <div
            class="sc-jRQBWg gDsAIL"
            data-testid="description"
          >
            <div
              aria-label="Meta NFT"
              class="ant-typography ant-typography-ellipsis sc-iqseJM gYJxwC"
            >
              Meta NFT
              <span
                aria-hidden="true"
                style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; word-break: keep-all; white-space: nowrap;"
              >
                lg
              </span>
              <span
                aria-hidden="true"
                style="position: fixed; display: block; left: 0px; top: 0px; z-index: -9999; visibility: hidden; pointer-events: none; width: 0px; white-space: normal; margin: 0px; padding: 0px;"
              >
                <span
                  aria-hidden="true"
                >
                  ...
                </span>
                <a
                  aria-label="Expand"
                  class="ant-typography-expand"
                >
                  Read More
                </a>
              </span>
            </div>
          </div>
          <div
            class="ant-space ant-space-horizontal ant-space-align-center sc-kDTinF hwEuWt"
          >
            <div
              class="ant-space-item"
              style="margin-right: 16px;"
            >
              <button
                class="ant-btn ant-btn-default sc-iCfMLu cUwNdV"
                type="button"
              >
                <span>
                  Sell
                </span>
              </button>
            </div>
            <div
              class="ant-space-item"
            >
              <div
                class="sc-pVTFL eZPElS ant-dropdown-trigger"
              >
                <span
                  class="sc-jrQzAO aGbEU"
                >
                  ...
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
